### PR TITLE
Fix VsphereMachineConfig Webhook validation on worker nodes

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

To allow worker nodes update on fields:
- folder
- resourcePool
- datastore
- diskGiB
- numCPUs
- memoryMiB
- template

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/395#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/v1alpha1/vspheremachineconfig_webhook.go [hunk 1]
```diff
@@ -91,6 +84,19 @@ func validateImmutableFieldsVSphereMachineConfig(new, old *VSphereMachineConfig)
 		)
 	}
 
+	if !old.IsControlPlane() && !old.IsEtcd() {
+		vspheremachineconfiglog.Info("Machine config is not associated with control plane or etcd")
+		return allErrs
+	}
+	vspheremachineconfiglog.Info("Machine config is associated with control plane or etcd")
+
+	if old.Spec.Template != new.Spec.Template {
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/vspheremachineconfig_webhook.go:93]: We probably should not make this mutable until we are able to auto import templates and ensure that this won't break worker node configuration?

**Comment by abhay-krishna** [comment]
New changes are detected. LGTM label has been removed.

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/395#" title="Author self-approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/395#pullrequestreview-772097869" title="LGTM">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**